bpfd/src/multiprog/xdp.rs

```diff
@@ -0,0 +1,221 @@
+            revision,
+            mode,
+            loader: Some(loader),
+        };
+        dispatcher.attach_extensions(&mut extensions)?;
+        dispatcher.attach()?;
+        dispatcher.save()?;
+        if let Some(old) = old_dispatcher {
+            old.delete(false)?;
+        }
+        Ok(dispatcher)
+    }
+
+    pub(crate) fn attach(&mut self) -> Result<(), BpfdError> {
+        let if_index = self.if_index;
+        let iface = self.if_name.clone();
+        let dispatcher: &mut Xdp = self
+            .loader
+            .as_mut()
+            .ok_or(BpfdError::NotLoaded)?
+            .program_mut(DISPATCHER_PROGRAM_NAME)
+            .unwrap()
+            .try_into()?;
+
+        let path = PathBuf::from(format!("{RTDIR_FS_XDP}/dispatcher_{if_index}_link"));
+        if path.exists() {
+            let pinned_link: FdLink = PinnedLink::from_pin(path).unwrap().into();
+            dispatcher
+                .attach_to_link(pinned_link.try_into().unwrap())
+                .unwrap();
+        } else {
+            let flags = self.mode.as_flags();
+            let link = dispatcher.attach(&iface, flags).unwrap();
+            let owned_link = dispatcher.take_link(link)?;
+            let path = format!("{RTDIR_FS_XDP}/dispatcher_{if_index}_link");
+            let _ = TryInto::<FdLink>::try_into(owned_link)
+                .unwrap() // TODO: Don't unwrap, although due to minimum kernel version this shouldn't ever panic
+                .pin(path)
+                .map_err(|_| BpfdError::UnableToPin)?;
+        }
+        Ok(())
+    }
+
+    fn attach_extensions(
+        &mut self,
+        extensions: &mut [(&Uuid, &XdpProgram)],
+    ) -> Result<(), BpfdError> {
+        let if_index = self.if_index;
+        let dispatcher: &mut Xdp = self
+            .loader
+            .as_mut()
+            .ok_or(BpfdError::NotLoaded)?
+            .program_mut(DISPATCHER_PROGRAM_NAME)
+            .unwrap()
+            .try_into()?;
+        extensions.sort_by(|(_, a), (_, b)| a.info.current_position.cmp(&b.info.current_position));
+        for (i, (k, v)) in extensions.iter_mut().enumerate() {
+            if v.info.metadata.attached {
+                let mut prog = PinnedProgram::from_pin(format!("{RTDIR_FS}/prog_{k}"))?;
+                let ext: &mut Extension = prog.as_mut().try_into()?;
+                let target_fn = format!("prog{i}");
+                let new_link_id = ext
+                    .attach_to_program(dispatcher.fd().unwrap(), &target_fn)
+                    .unwrap();
+                let new_link: FdLink = ext.take_link(new_link_id)?.into();
+                let path = format!(
+                    "{RTDIR_FS_XDP}/dispatcher_{if_index}_{}/link_{k}",
+                    self.revision
+                );
+                new_link.pin(path).map_err(|_| BpfdError::UnableToPin)?;
+            } else {
+                let mut bpf = BpfLoader::new()
+                    .extension(&v.data.section_name)
```

Comment:
```suggestion
                    .extension(&v.data.section_name)
                    .map_pin_path(map_pin_path)
```
Except need to pull from v.data